src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,50 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int [] diagonal = new int[matrix[0].length];
+        for (int i = 0; i < matrix[0].length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int [] counterDiagonal = new int[matrix[0].length];
+        int counter = matrix[0].length - 1;
```

Comment:
do we need this variable?

Reply:
I have somehow to get second coordinate to move from right to left. Is it possible to do it without this variable?

Reply:
I meant do you need to create a variable for it if you are using it only once and it's short enough to write it as it is.
Think about getting this value from i.

Reply:
Fixed this issue. Thanks for help :)